projects/observability/src/shared/services/csv-downloader/csv-downloader.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Dictionary } from '@hypertrace/common';
+import { CsvDownloadFileConfig, FileDownloadService } from '@hypertrace/components';
+import { Observable, of } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class CSVDownloaderService {
+  private readonly componentHandlers: Map<string, Observable<Dictionary<unknown>[]>> = new Map();
```

Comment:
instead of handler, we should use dataSource everywhere, that would be a more appropriate name.
`componentHandlers ` -> `csvDataSourceMap`
`registerHandler ` -> `registerDataSourceMap`
`hasRegisteredHandler ` -> `hasRegisteredDataSource`
`deleteRegisteredHandler ` -> `deleteRegisteredDataSource`
`clearRegisteredHandler ` -> `clearAllDataSource`

Reply:
done